File: lua/8-structures.lua
#!/bin/lua
car={}
brand=1
year=2

function show(s)
    print("brand:", s["brand"], s[brand])
    print("year of production:", s["year"], s[year])
end

car["brand"]="volvo v40"
car["year"]=2010

car[brand]="volvo v40"
car[year]=2010
    
show(car)

File: perl/8-structures.pl
#!/bin/perl

%volvo=( 
	'brand' => 'volvo v40', 
	'year_of_production' => 2010);

sub show
{
	my (%h) = @_;
	print "brand: $h{'brand'}\n";
	print "year of production: $h{'year_of_production'}\n";

}

sub hash_tab 
{
	my %h = @_;
	foreach my $key (keys %h) {
		print "key=$key value=$h{$key}\n";
	}
}


show(%volvo);
hash_tab(%volvo);
